### PR TITLE
Bugfix/remove custom selector schema settings from modules

### PR DESCRIPTION
**Pull request by thilo-behnke**
### Checklist
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~I documented my source code using the JavaDoc / JSDoc style.~
- [x] ~I added integration test cases for the server (Spring) related to the features~
- [x] ~I added integration test cases for the client (Jest) related to the features~
- [x] ~I added screenshots/screencast of my UI changes~
- [x] ~I translated all the newly inserted strings~

### Motivation and Context

In many module definitions we use the following setting:
```
@ngModule({
...
schemas: [CUSTOM_ELEMENTS_SCHEMA]
...
)}
```
(https://angular.io/api/core/CUSTOM_ELEMENTS_SCHEMA)

:information_source: This setting has harmful effects as we might not detect bugs in the template (see examples below and screenshot).

### Description

The setting mentioned above allows e.g. the following html without `code-error` being a valid selector for an angular component:

```
<code-error>some random text</code-error>
```

**This setting causes a couple of issues:**
- When a component from another module is not imported, no error is raised by angular, but an empty html tag is rendered (see screenshot below for an example)
- No wrong inputs/outputs are recognized (e.g. if the component does not have an input)

### Steps for Testing

**When allowing custom schemas (current state):**
- Check out the current version of develop
- Comment out the `ArtemisResultsModule` import in `code-editor.module.ts`
- Load the code-editor in the client => the result will not be shown as angular rendered an empty html tag for the `jhi-updating-result`

**When NOT allowing custom schemas (new state):**
- Check out this branch
- Comment out the `ArtemisResultsModule` import in `code-editor.module.ts`
- When trying to access the client, an error will be shown: `Uncaught Error: Template parse errors:
Can't bind to 'exerciseType' since it isn't a known property of 'jhi-updating-result'` (This is the expected result from angular)

### Screenshots

**Current state of develop when a component import is missing:**

image

### CODE REVIEW

File: src/main/webapp/app/entities/course/course-update.component.html [hunk 1]
```diff
@@ -62,11 +62,10 @@ <h4 *ngIf="course.id" id="jhi-course-heading-edit" jhiTranslate="artemisApp.cour
                             <image-cropper
                                 [imageChangedEvent]="imageChangedEvent"
                                 [maintainAspectRatio]="true"
-                                [aspectRatio]="1 / 1"
+                                [aspectRatio]="1"
                                 [resizeToWidth]="200"
                                 [cropperMinWidth]="100"
-                                [cropperMaxWidth]="400"
```
Comment (thilo-behnke) [inline src/main/webapp/app/entities/course/course-update.component.html]: This setting does not exist and never could have done anything.

File: src/main/webapp/app/entities/course/course-update.component.html [hunk 2]
```diff
@@ -62,11 +62,10 @@ <h4 *ngIf="course.id" id="jhi-course-heading-edit" jhiTranslate="artemisApp.cour
...
-                                [lyScaleDown]="true"
```
Comment (thilo-behnke) [inline src/main/webapp/app/entities/course/course-update.component.html]: Typo that was not detected.

File: src/main/webapp/app/entities/result/result-detail.component.html [hunk 3]
```diff
@@ -8,17 +8,17 @@ <h4 class="modal-title">Feedback</h4>
     </div>
     <div [hidden]="isLoading" *ngFor="let feedback of feedbackList | removepositiveautomaticfeedback; last as isLast">
         <p *ngIf="feedback.type === 'AUTOMATIC' && feedback.positive === false">
-            Error in &nbsp;<code-error>{{ feedback.text }}</code-error
+            Error in &nbsp;<span class="code-error">{{ feedback.text }}</span
```
Comment (thilo-behnke) [inline src/main/webapp/app/entities/result/result-detail.component.html]: There is no reason for doing this as we could just a css class.

File: src/main/webapp/app/shared/image/secured-image.component.ts [hunk 4]
```diff
@@ -44,6 +44,7 @@ export class SecuredImageComponent implements OnChanges {
     @Input() mobileDragAndDrop: boolean;
     @Input() private src: string;
     @Input() cachingStrategy = CachingStrategy.SESSION_STORAGE;
+    @Input() alt = '';
```
Comment (thilo-behnke) [inline src/main/webapp/app/shared/image/secured-image.component.ts]: This input was missing, so the alt tag was not displayed (not a major issue). However this is a problem of the schema setting as the compiler did not detect the issue.

**Comment by krusche** [comment]
CUSTOM_ELEMENTS_SCHEMA comes from the jhipster template. I don't know why they actually use it and what it is good for.

From my side, we can remove CUSTOM_ELEMENTS_SCHEMA if it improves our compile and build process and if we can detect more errors. 

We might even report this to the jhipster developers and discuss why they use it.

**Comment by krusche** [comment]
@thilo-behnke Did you remove CUSTOM_ELEMENTS_SCHEMA from all modules in this PR?

**Comment by thilo-behnke** [comment]
> @thilo-behnke Did you remove CUSTOM_ELEMENTS_SCHEMA from all modules in this PR?

Yes, I removed it from all modules.

**Comment by thilo-behnke** [comment]
I had to make a small css fix, this shouldn't break the build - ready for review.

**Pull request merged**